diff -r 469d87c15a5b CMakeLists.txt
--- a/CMakeLists.txt	Tue May 29 16:53:11 2018 -0700
+++ b/CMakeLists.txt	Mon Jun 04 17:48:39 2018 +1000
@@ -67,6 +67,11 @@
     cmake_policy(SET CMP0060 NEW)
 endif()
 
+# Make IMPORTED_LINK_INTERFACE_LIBRARIES per configuration when exported
+if(POLICY CMP0022)
+   cmake_policy(SET CMP0022 OLD)
+endif()
+
 
 #---------------------------------------------------------------------
 # Set project information
@@ -1910,7 +1921,7 @@
         ${BIM_LIB_EXIV2}/exiv2/utilsvideo.hpp)
 
     internal_add_library(exiv2 "${BIC_INTERNAL_EXIV2_SOURCES}" "${BIC_INTERNAL_EXIV2_HEADERS}" "${BIC_INTERNAL_EXIV2_INSTALLHEADERS}" "${BIC_INTERNAL_EXIV2_LINK_LIBRARIES}")
-    target_compile_definitions(exiv2 PUBLIC SUPPRESS_WARNINGS _FILE_OFFSET_BITS=64 EXV_BUILDING_LIB=1 DLL_EXPORT)
+    target_compile_definitions(exiv2 PUBLIC SUPPRESS_WARNINGS _FILE_OFFSET_BITS=64 EXV_BUILDING_LIB=1)
 
     set(EXIV2_LIBRARIES "${LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_LIBRARY_PREFIX}exiv2${CMAKE_LIBRARY_SUFFIX}")
 elseif(LIBBIOIMAGE)
diff -r 469d87c15a5b libsrc/libbioimg/formats/bim_exiv_parse.cpp
--- a/libsrc/libbioimg/formats/bim_exiv_parse.cpp	Tue May 29 16:53:11 2018 -0700
+++ b/libsrc/libbioimg/formats/bim_exiv_parse.cpp	Mon Jun 04 17:48:39 2018 +1000
@@ -107,13 +107,7 @@
     Exiv2::Image::AutoPtr image;
     if (!fmtHndl->fileName) return image;
     try {
-#ifdef BIM_WIN
-        bim::xstring fn(fmtHndl->fileName);
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(fn.toUTF16());
-#else
         Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(fmtHndl->fileName);
-#endif
-
         if (image.get() == 0) return Exiv2::Image::AutoPtr();
         image->readMetadata();
         return image;
@@ -203,13 +197,7 @@
         Exiv2::ExifParser::decode(exifData,
             (const Exiv2::byte *) hash->get_value_bin(bim::RAW_TAGS_EXIF),
             hash->get_size(bim::RAW_TAGS_EXIF));
-            
-#ifdef BIM_WIN
-        bim::xstring fn(fmtHndl->fileName);
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(fn.toUTF16());
-#else
         Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(fmtHndl->fileName);
-#endif
         if (image.get() == 0) return;
         image->readMetadata();
         image->setExifData(exifData);