diff -r 6a9599d65fa2 CMakeLists.txt
--- a/CMakeLists.txt	Tue Jun 05 09:58:53 2018 +1000
+++ b/CMakeLists.txt	Thu Jun 07 13:51:07 2018 +1000
@@ -67,6 +67,11 @@
     cmake_policy(SET CMP0060 NEW)
 endif()
 
+# Make IMPORTED_LINK_INTERFACE_LIBRARIES per configuration when exported
+if(POLICY CMP0022)
+   cmake_policy(SET CMP0022 OLD)
+endif()
+
 
 #---------------------------------------------------------------------
 # Set project information